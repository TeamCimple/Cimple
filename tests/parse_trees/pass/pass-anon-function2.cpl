int apply_int_op(int param, func(int)(int) op)
{
	return op(param);
}

int main()
{
	int a = 1;
	int b = apply_int_op(1, func(int)(int x) { return a + x;});
	return apply_int_op(b, func(int)(int x) { return x - 2; });
}
