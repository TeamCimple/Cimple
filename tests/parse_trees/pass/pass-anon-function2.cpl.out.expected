FuncDecl(Name: Var(main) ReturnType: int Parameters:  Body: CompoundStatement(Declarations: (int EQUALS(Var(a) Int(1),), (int EQUALS(Var(b) Call(Receiver()FunctionName: Identifier(apply_int_op) Params: AnonFuncDef(ReturnType: int, Params: PARAM(int Var(x)) , Body: CompoundStatement(Declarations:  StatementList: RETURN(PLUS(Identifier(a), Identifier(x)))))Int(1)),),  StatementList: RETURN(Call(Receiver()FunctionName: Identifier(apply_int_op) Params: AnonFuncDef(ReturnType: int, Params: PARAM(int Var(x)) , Body: CompoundStatement(Declarations:  StatementList: RETURN(MINUS(Identifier(x), Int(2)))))Identifier(b))))) , FuncDecl(Name: Var(apply_int_op) ReturnType: int Parameters: AnonFuncDecl(Name: op, ReturnType: int, Params: PARAM(int) ), PARAM(int Var(param))  Body: CompoundStatement(Declarations:  StatementList: RETURN(Call(Receiver()FunctionName: Identifier(op) Params: Identifier(param))))) 
